### PR TITLE
tikv: add function SplitRegionRanges to get the split ranges from region.

### PR DESCRIPTION
**Pull request by winkyao**
The SplitRegionRanges function will be used in adding index write-reorganization stage.

@zimulala @coocood @shenli  PTAL

### CODE REVIEW

**Comment by winkyao** [comment]
@disksing PTAL

File: store/tikv/coprocessor.go [hunk 1]
```diff
@@ -315,19 +331,31 @@ func buildCopTasks(bo *Backoffer, cache *RegionCache, ranges *copRanges, desc bo
...
-		reverseTasks(tasks)
+	return nil
+}
+
+// SplitRegionRanges get the split ranges from pd region.
+func SplitRegionRanges(bo *Backoffer, cache *RegionCache, keyRanges []kv.KeyRange) ([]kv.KeyRange, error) {
+	ranges := copRanges{mid: keyRanges}
+
+	var ret []kv.KeyRange
+	appendRange := func(region RegionVerID, ranges *copRanges) {
```
Comment (shenli) [inline store/tikv/coprocessor.go]: What is this region used for?

**Comment by disksing** [comment]
LGTM.

**Comment by shenli** [comment]
LGTM

File: store/tikv/coprocessor.go [hunk 2]
```diff
@@ -315,19 +331,31 @@ func buildCopTasks(bo *Backoffer, cache *RegionCache, ranges *copRanges, desc bo
...
-	if desc {
-		reverseTasks(tasks)
+	return nil
+}
+
+// SplitRegionRanges get the split ranges from pd region.
+func SplitRegionRanges(bo *Backoffer, cache *RegionCache, keyRanges []kv.KeyRange) ([]kv.KeyRange, error) {
+	ranges := copRanges{mid: keyRanges}
+
+	var ret []kv.KeyRange
```
Comment (ngaut) [inline store/tikv/coprocessor.go]: Make with capacity.

**Pull request merged**